From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 27 Oct 2025 13:53:11 -0400
Subject: [PATCH] build: Add option to disable tests

We currently need this for Solus because we're building packages as root
because fakeroot is broken with our setup or something, I don't remember
the details.

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build       | 4 +++-
 meson_options.txt | 1 +
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4a509e7..845f7f3 100644
--- a/meson.build
+++ b/meson.build
@@ -232,7 +232,9 @@ if get_option('gtk_doc')
   subdir('doc/libaccountsservice')
 endif
 
-subdir('tests')
+if get_option('tests') and meson.can_run_host_binaries()
+  subdir('tests')
+endif
 
 configure_file(
   output: 'config.h',
diff --git a/meson_options.txt b/meson_options.txt
index b34a0fa..67db09b 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -13,3 +13,4 @@ option('vapi', type: 'boolean', value: true, description : 'Enable Vala bindings
 
 option('docbook', type: 'boolean', value: false, description: 'build documentation (requires xmlto)')
 option('gtk_doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
+option('tests',   type: 'boolean', value: true, description: 'run accountservice tests if possible')
